core/src/main/java/com/graphhopper/routing/util/BikeCommonFlagEncoder.java

```diff
@@ -559,7 +574,10 @@ void collect( OSMWay way, double wayTypeSpeed, TreeMap<Double, Integer> weightTo
         }
         
         if ("cycleway".equals(highway))
-            weightToPrioMap.put(100d, VERY_NICE.getValue());
+            if (way.hasTag("foot", intendedValues) && !way.hasTag("segregated", "yes"))
```

Comment:
Can you put brakets around the parent if to make the flow clear (although logic should be the same)?